cmd/access/node_builder/staked_access_node_builder.go

```diff
@@ -272,60 +284,88 @@ func (anb *StakedAccessNodeBuilder) Build() (cmd.Node, error) {
 					// since we are always more up-to-date than them
 					false,
 				)
-
 				// register the sync request handler with the proxy engine
 				proxyEngine.RegisterEngine(syncRequestHandler)
 
 				return syncRequestHandler, nil
+			}).
+			Component("relay engine", func(node *cmd.NodeConfig) (module.ReadyDoneAware, error) {
+				return relay.New(
+					node.Logger,
+					network.ChannelList{
+						engine.ReceiveBlocks,
+					},
+					node.Network,
+					anb.AccessNodeConfig.PublicNetworkConfig.Network,
+				)
 			})
 	}
 
 	anb.Component("ping engine", func(node *cmd.NodeConfig) (module.ReadyDoneAware, error) {
+		// setup the Ping provider to return the software version and the sealed block height
+		pingProvider := p2p.PingInfoProviderImpl{
+			SoftwareVersionFun: func() string {
+				return build.Semver()
+			},
+			SealedBlockHeightFun: func() (uint64, error) {
+				head, err := node.State.Sealed().Head()
+				if err != nil {
+					return 0, err
+				}
+				return head.Height, nil
+			},
+			HotstuffViewFun: func() (uint64, error) {
+				return 0, fmt.Errorf("non-consensus nodes do not report hotstuff view in ping")
+			},
+		}
+
+		pingLibP2PProtocolID := unicast.PingProtocolId(node.SporkID)
+		pingService := p2p.NewPingService(node.Middleware.Host(), pingLibP2PProtocolID, pingProvider, node.Logger)
+
```

Comment:
At first I thought this makes sense, but I just went back and looked at the code now, and I realized the reason ping service is initialized outside of ping engine is because it is needed on its own for non-Access nodes.

The ping engine is used by staked ANs to ping all other nodes, and all other nodes use the PingService to respond to ping requests. So all other node types simply use PingService component without the PingEngine wrapping it.

However, I think it probably does make sense to move the ping provider initialization into the pingservice.

Reply:
fc511f6